### PR TITLE
Bump ocamlformat version from 0.18.0 to 0.19.0

### PR DESCRIPTION
**Pull request by arozovyk**
This PR is related to #666, since it introduces `ppxlib`. 

This change will be needed since `ocamlformat.0.18.0` used `ocaml-migrate-parsetree` in pair with `ppxlib`s' internal API (`Ppxlib_ast__.Versions.OCaml_408`) that will change in the the upcoming release of `ppxlib` making it incompatible. 

The `0.19.0` [drops](https://github.com/ocaml-ppx/ocamlformat/commit/448ac7c1496e2a6e83d63a7bcd9cf4e35bb2b1fb) the dependency and is therefore compatible.

### CODE REVIEW

**Review by mnxn — APPROVED** [review]

**Comment by ulugbekna** [comment]
1. I recently learned about `git blame --ignore-rev <commit-hash>` which allows to have good git blame, by ignoring commits that have bulk changes such as 81b132ce6b5b54c611c709fdc085d470b6c9abf2 in this case. What do you think about having `.git-blame-ignore-revs` file with the following contents? 

```
# upgrade to ocamlformat.0.19.0
81b132ce6b5b54c611c709fdc085d470b6c9abf2
```

Then, one can do `git blame --ignore-revs-file .git-blame-ignore-revs` to have correct git blames :-) This ignores `make fmt` commit, for example:

image

(I don't insist however)

2. Before this is merged, I think it's good to squash commits. And if `.git-blame-ignore-revs` used, update the commit hash.

**Comment by smorimoto** [comment]
I thought it was good in the old time, but when do you add a commit to it? VSCode's format integration easily reformats the code "while" you are writing the code. I used to do that but I don't do that intentionally these days because strict blaming is impossible. I don't have a strong opinion on this either, but it's very hard to figure out which commit to add to it and which not. If someone has a good idea for it, I want to use it again on my project as well.

**Comment by ulugbekna** [comment]
Getting different formatting when you add code is normal I think. One wants
to avoid commits that are "formatting" only. Such commits occur when one
upgrades ocamlformat, which isn't stable yet and slightly changes
formatting with every version

**Comment by ulugbekna** [comment]
This should be good to go, but I'm holding off merging it until we bump
ocamlformat in ocaml-lsp as well (to keep these projects in the same
switch). @rgrinberg, I think you prefer to have ocamlformat version bumped
in ocaml-lsp at the same time as in dune. Do you know when this is
happening? (I can send PRs)

**Comment by rgrinberg** [comment]
yes, feel free to upgrade to 0.19

**Pull request merged**
